tonic/src/transport/service/reconnect.rs

```diff
@@ -52,6 +54,7 @@ where
     M::Future: Unpin,
     Error: From<M::Error> + From<S::Error>,
     Target: Clone,
+    <M as tower_service::Service<Target>>::Error: fmt::Debug,
```

Comment:
This still needs to be `Into<Error>` or else it might break the stack down the line.